src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,582 @@
+            cluster=Clusters.BasicInformation,
+            attribute=Clusters.Attribute.DataVersion
+        )
+
+        # Get DataVersion
+        data_version = sub_cr1_empty_dvf[0][Clusters.BasicInformation][Clusters.Attribute.DataVersion]
+        data_version_filter = [(0, Clusters.BasicInformation, data_version)]
+
+        # Subscribe to attribute with provided DataVersion
+        sub_cr1_provided_dvf = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=node_label_attr_path,
+            reportInterval=(10, 20),
+            keepSubscriptions=False,
+            dataVersionFilters=data_version_filter
+        )
+
+        # Verify that the subscription is activated between CR1 and DUT
+        asserts.assert_true(sub_cr1_provided_dvf.subscriptionId, "Subscription not activated")
+
+        sub_cr1_provided_dvf.Shutdown()
+
+        '''
+        ##########
+        Step 8
+        ##########
+        '''
+        self.print_step(8, "CR1 sends a subscription request action for an attribute and sets the MinIntervalFloor value to be same as MaxIntervalCeiling. Activate the Subscription between CR1 and DUT. Modify the attribute which has been subscribed to on the DUT.")
+
+        # Subscribe to attribute
+        same_min_max_interval_sec = 3
+        sub_cr1_update_value = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=node_label_attr_path,
+            reportInterval=(same_min_max_interval_sec, same_min_max_interval_sec),
+            keepSubscriptions=False
+        )
+
+        # Modify attribute value
+        new_node_label_write = "NewNodeLabel_011235813"
+        await CR1.WriteAttribute(
+            self.dut_node_id,
+            [(0, node_label_attr(value=new_node_label_write))]
+        )
+
+        # Wait MinIntervalFloor seconds before reading updated attribute value
+        time.sleep(same_min_max_interval_sec)
+        new_node_label_read = sub_cr1_update_value.GetAttribute(node_label_attr_typed_path)
+
+        # Verify new attribute value after MinIntervalFloor time
+        asserts.assert_equal(new_node_label_read, new_node_label_write, "Attribute value not updated after write operation.")
+
+        sub_cr1_update_value.Shutdown()
+
+        '''
+        ##########
+        Step 9
+        ##########
+        '''
+        self.print_step(
+            9, "CR1 sends a subscription request action for an attribute and set the MinIntervalFloor value to be greater than MaxIntervalCeiling.")
+
+        # Subscribe to attribute with invalid reportInterval arguments, expect and exception
+        sub_cr1_invalid_intervals = None
+        try:
+            sub_cr1_invalid_intervals = await CR1.ReadAttribute(
+                nodeid=self.dut_node_id,
+                attributes=node_label_attr_path,
+                reportInterval=(20, 10),
+                keepSubscriptions=False
+            )
+        except ChipStackError:
```

Comment:
add a check for try not failing as expected here as well.

Reply:
@raul-marquez-csa please add as a followup. Approving to merge for now since this is good to have for te2.